src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs

```diff
@@ -889,10 +915,32 @@ public unsafe void ServiceMainCallback(int argCount, IntPtr argPointer)
 
         private void WriteLogEntry(string message, bool error = false)
         {
-            // Used to write to EventLog but for now just logging to debug output stream
-            // might want to plumb other logging in the future.
-
-            Debug.WriteLine((error ? "Error: " : "") + message);
+            // EventLog failures shouldn't affect the service operation
+            try 
+            {
+                if (AutoLog)
+                {
+                    EventLog.WriteEntry(message); 
+                }
+            }
+            #region Stuff not to catch
```

Comment:
nit: normally I would avoid using region directives in code unless it's already present in the code style of the class I'm making changes to. 

Please take a look at [this link](https://softwareengineering.stackexchange.com/questions/53086/are-regions-an-antipattern-or-code-smell) for more details.